sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareDirectoryAsyncClient.java

```diff
@@ -455,14 +454,15 @@ Mono<Response<ShareDirectoryInfo>> createIfNotExistsWithResponse(ShareDirectoryC
         Context context) {
         try {
             options = options == null ? new ShareDirectoryCreateOptions() : options;
-            return createWithResponse(options.getSmbProperties(), options.getFilePermission(), null,
-                options.getMetadata(), context).onErrorResume(
-                    t -> t instanceof ShareStorageException && ((ShareStorageException) t).getStatusCode() == 409,
-                    t -> {
-                        HttpResponse response = ((ShareStorageException) t).getResponse();
-                        return Mono.just(new SimpleResponse<>(response.getRequest(), response.getStatusCode(),
-                            response.getHeaders(), null));
-                    });
+            return createWithResponse(options.getSmbProperties(), options.getFilePermission(),
+                options.getFilePermissionFormat(), options.getPosixProperties(), options.getMetadata(), context)
```

Comment:
i added the feature that introduced filePermission/filePermissionFormat a while ago - i missed adding the file permission format here at that time, so i figured i'd add it now before we got any issues about it haha

Reply:
Gotcha. Yeah I'm fine with calling it a bug fix assuming that if the customer did not set the file permission format to start it would have resolved to `null`. It would only be an issue if there was a default format the options bag was explicitly resolving to.